FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```
@@ -19,6 +19,41 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 	
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		
+		self.client.get(from: self.url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case let .success((data, response)):
+				switch response.statusCode {
+				case 200:
+					if let images = FeedImageMapper.feedImages(from: data) {
+						completion(.success(images))
+					} else {
+						completion(.failure(Error.invalidData))
+					}
+				default:
+					completion(.failure(Error.invalidData))
+				}
+			}
+		}
+	}
+}
+
+private struct FeedImageMapper {
+	static func feedImages(from data: Data) -> [FeedImage]? {
+		guard let root = try? JSONSerialization.jsonObject(with: data) as? [String: [[String: String]]],
+			  let items = root["items"]
+		else { return nil }
+
+		return items.compactMap {
+			guard let uuIdString = $0["image_id"],
+				  let uuId = UUID(uuidString: uuIdString),
+				  let urlString = $0["image_url"],
+				  let url = URL(string: urlString)
+			else { return nil }
```

It's important to notice that `compactMap` will discard any `nil` results. 

A `nil` result in this context means invalid data (an item with missing or invalid id/URL).

And if the mapping of any item fails for any reason, it should fail the whole operation because that'd be an invalid response (not following the API contract).

An expressive way of dealing with this failure would be to mark the `FeedImageMapper.feedImages` method with `throws` instead of returning optional `[FeedImage]?`.

If the mapping fails for any reason, it can throw `.invalidData`.